packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -70,16 +72,22 @@ export function onBlockDrop(
 	getClientIdsOfDescendants,
 	moveBlocks,
 	insertOrReplaceBlocks,
-	clearSelectedBlock
+	clearSelectedBlock,
+	context
 ) {
 	return ( event ) => {
 		const {
 			srcRootClientId: sourceRootClientId,
 			srcClientIds: sourceClientIds,
 			type: dropType,
 			blocks,
+			context: dragContext,
 		} = parseDropEvent( event );
 
+		if ( context && dragContext && context !== dragContext ) {
+			return;
+		}
```

Comment:
Needs a comment. Basically if 

- you defined a drag and drop context 
- where you started the drag doesn't match where you ended it

...then disallow that drag and drop.